src/dali/plugin/input/csv/binance_com.py

```diff
@@ -0,0 +1,175 @@
+    __AUTO_BASE_SYMBOL: int = 1
+    __AUTO_QUOTE_AMOUNT_SYMBOL: int = 2
+    __AUTO_TRADING_FEE_SYMBOL: int = 3
+    __BETHETH_AMOUNT: int = 3
+    __AUTO_BASE_AMOUNT_SYMBOL: int = 4
+    __FUND_SOURCE: int = 5
+
+    __DELIMITER: str = ","
+
+    def __init__(
+        self,
+        account_holder: str,
+        autoinvest_csv_file: Optional[str] = None,
+        betheth_csv_file: Optional[str] = None,
+        native_fiat: Optional[str] = None,
+    ) -> None:
+
+        super().__init__(account_holder=account_holder, native_fiat=native_fiat)
+        self.__autoinvest_csv_file: Optional[str] = autoinvest_csv_file
+        self.__betheth_csv_file: Optional[str] = betheth_csv_file
+        self.__logger: logging.Logger = create_logger(f"{self.__BINANCE_COM_CSV}/{self.account_holder}")
+
+    def load(self) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        if self.__autoinvest_csv_file:
+            result = self.parse_autoinvest_file(self.__autoinvest_csv_file)
+
+        if self.__betheth_csv_file:
+            if result:
+                result += self.parse_betheth_file(self.__betheth_csv_file)
+            else:
+                result = self.parse_betheth_file(self.__betheth_csv_file)
+
+        return result
+
+    def parse_autoinvest_file(self, file_path: str) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        with open(file_path, encoding="utf-8") as csv_file:
+            lines = reader(csv_file)
+
+            header = next(lines)
+            self.__logger.debug("Header: %s", header)
+            for line in lines:
+                raw_data: str = self.__DELIMITER.join(line)
+                self.__logger.debug("Transaction: %s", raw_data)
+
+                result.append(
+                    InTransaction(
+                        plugin=self.__BINANCE_COM_CSV,
+                        unique_id=Keyword.UNKNOWN.value,
+                        raw_data=raw_data,
+                        timestamp=f"{line[self.__TIMESTAMP_INDEX]} -00:00",
+                        asset=line[self.__AUTO_BASE_SYMBOL],
+                        exchange=self.__BINANCE_COM,
+                        holder=self.account_holder,
+                        transaction_type=Keyword.BUY.value,
+                        spot_price=Keyword.UNKNOWN.value,
+                        crypto_in=(line[self.__AUTO_BASE_AMOUNT_SYMBOL]).split()[0],
+                        notes=f"Autoinvest buy with funding from {line[self.__FUND_SOURCE]}",
+                    )
+                )
+
+                quote_asset_symbol: str = line[self.__AUTO_QUOTE_AMOUNT_SYMBOL].split()[1]
+                quote_asset_amount: str = line[self.__AUTO_QUOTE_AMOUNT_SYMBOL].split()[0]
+                crypto_fee: str = line[self.__AUTO_TRADING_FEE_SYMBOL].split()[0]
+                crypto_out_with_fee: RP2Decimal = RP2Decimal(quote_asset_amount) + RP2Decimal(crypto_fee)
+                result.append(
```

Comment:
We're appending two same-timestamp transactions per loop (a in- and an out-transaction): could you educate me on how this model works?

Reply:
Autoinvest uses a stable coin (BUSD, USDT, USDC) to automatically invest in a crypto asset at a set time each day/week/biweek/month via swap. Each one of these transactions is a conversion. (e.g. 2.00 USDT is converted into 0.0001 BTC.)